cpu/stm32/stmclk/stmclk_common.c

```diff
@@ -106,5 +106,10 @@ void stmclk_dbp_unlock(void)
 
 void stmclk_dbp_lock(void)
 {
-    PWR->REG_PWR_CR &= ~(BIT_CR_DBP);
+#if defined(CPU_HAS_BACKUP_RAM)
+/*  Unless we go to sleep, the DBP must be unlocked all the time, if we modify
+    backup RAM content by comfortable BACKUP_RAM variables */
+    if (backup_ram_is_retained())
+#endif
+        PWR->REG_PWR_CR &= ~(BIT_CR_DBP);
```

Comment:
```suggestion
    if (!IS_ACTIVE(CPU_HAS_BACKUP_RAM) || backup_ram_is_retained()) {
        PWR->REG_PWR_CR &= ~BIT_CR_DBP;
    }
```

Reply:
Then the API `backup_ram_*` in `periph_cpu.h` should always be visible?

Reply:
Sure, what speaks against it?
That's certainly better than such wacky preprocessor constructs